README.md

```diff
@@ -0,0 +1,216 @@
+# Python Aho-Corasick implementation
+
+`noahong` is a Python implementation of the [Aho-Corasick](https://en.wikipedia.org/wiki/Aho%E2%80%93Corasick_algorithm) algorithm for string matching, based on a fork of the [NoAho](https://github.com/JDonner/NoAho) C++ implementation.
+
+## API
+
+The first thing to do is to instantiate a `NoAho` object and add some keys to it (optionally with different payloads for each).
+
+```
+from noahong import NoAho
+
+trie = NoAho()
+
+# fill with .add()
+trie.add("foo", "id_foo")
+trie.add("foobar", "id_foobar")
+
+# or fill with __setitem__
+trie["bar"] = "id_bar"
+```
+
+Once you have added the different keys and their payloads, the `NoAho` object needs to be compiled:
+
+```
+trie.compile()
+```
+
+Once it is compiled, keys can no longer be added to the `NoAho` object.
+
+`noahong` then exposes four functions to find matching substrings in text:
+
+### `find_short`
+
+`trie.find_short(text)` finds the first substring of `text` that is matched by a key added to the `trie`. 
+
+It returns a tuple `(start, stop, payload)` such that: 
+- `payload` is the object inserted with `trie.add()`
+- `start` and `stop` are indices of the match in the `text`: `text[start:stop] == key` 
+
+For example, using the above `trie`:
+
+```
+trie.find_short("something foo")
+# returns (10, 13, 'id_foo')
```

Comment:
What do you think about the way the original README presented examples? I think it's safe to assume the reader as seen a Python REPL

```suggestion
>>> trie.find_short("something foo")
(10, 13, 'id_foo')
```

Reply:
Call me lazy, but I always like to be able to just copy/paste code into my REPL and see what happens, which typically the `>>>` prevent. Although they do in fact look much better.